src/ajax.js

```diff
@@ -559,7 +559,7 @@ jQuery.extend( {
 
 		// Convert data if not already a string
 		if ( s.data && s.processData && typeof s.data !== "string" ) {
-			s.data = jQuery.param( s.data, s.traditional );
+			s.data = jQuery.param( s.data, s.traditional, s );
```

Comment:
That is because `jQuery.param()` should not need to know that it is being called by `jQuery.ajax()`, it can be called by any code that wants to serialize a parameter using HTML form rules. This change is using `jQuery.param()` as a hook to an AJAX call for a totally unrelated thing. 

Can you do this by setting `processData: false` and using a `beforeSend` callback to format the data?